### ISSUE TITLE
Collectd ceph plugin fails when ceph is not running

### ISSUE THREAD

**Issue by rubenk** [collectd/collectd#3260]
*   Version of collectd: 5.8.1
*   Operating system / distribution: centos 7
*   Kernel version (if applicable):

Aug 21 12:58:29 ams1-ceph02-mon01 systemd[1]: Started Collectd statistics daemon.
Aug 21 12:58:29 ams1-ceph02-mon01 collectd[1654]: [error] ceph plugin: cconn_connect: connect(3) failed: error -2
Aug 21 12:58:29 ams1-ceph02-mon01 collectd[1654]: [warning] ceph plugin: cconn_prepare(name=ams1-ceph02-mon01-mgr,i=0,st=0)=-2
Aug 21 12:58:29 ams1-ceph02-mon01 collectd[1654]: [error] ceph plugin: cconn_connect: connect(0) failed: error -2
Aug 21 12:58:29 ams1-ceph02-mon01 collectd[1654]: [warning] ceph plugin: cconn_prepare(name=ams1-ceph02-mon01-mon,i=1,st=0)=-2
Aug 21 12:58:29 ams1-ceph02-mon01 collectd[1654]: [info] Initialization complete, entering read-loop.

Collectd does continue but never connects to Ceph again.

**Comment by mrunge** [comment]
Yes! It would be great if the plugin would retry to connect. The same will happen if one is restarting ceph and it takes some time to start up.
